src/header/common/referer.rs

```diff
@@ -0,0 +1,35 @@
+use header::{Header, HeaderFormat};
+use std::fmt::{self, Show};
+use header::shared::util::from_one_raw_str;
+
+/// The `Referer` header.
+///
+/// The Referer header is used by user agents to inform server about
+/// the page URL user has came from.
+///
+/// See alse [RFC 1945, section 10.13](http://tools.ietf.org/html/rfc1945#section-10.13).
+///
+/// Currently just a string, but maybe better replace it with url::Url or something like it.
```

Comment:
Actually it would be a good idea if header parsers could access the request to know the base URL. Like that the relative URL could be resolved immediately and we could use `url::Url`